src: restructure modules and bootstrappers

joyeecheung opened the pull request:
<!--
Thank you for your pull request. Please provide a description above and review
the requirements below.

Bug fixes and new features should include tests and possibly benchmarks.

Contributors guide: https://github.com/nodejs/node/blob/master/CONTRIBUTING.md
-->

Refs: https://github.com/nodejs/node/pull/19112

### src: put bootstrappers in lib/internal/bootstrap/

Create `lib/internal/bootstrap/` and put bootstrappers there:

Before:

```
lib/internal
├── ...
├── bootstrap_loaders.js
└── bootstrap_node.js
```

After:

```
lib/internal
├── ...
└── bootstrap
    ├── loaders.js
    └── node.js
```

### lib: restructure cjsm and esm loaders

Create `lib/internal/modules` and restructure the module loaders
to make the purpose of those files clearer.

Also make it clear in the code that the object exported by
`lib/internal/modules/cjsm.js` is `CJSModule` instead of the
ambiguous `Module`.

Before (three files named `module.js` without any indication of
what they are implementing):

```
lib
├── ...
├── internal
│       ├── process
│       │     └── module.js
│       ├── loaders
│       │     ├── CreateDynamicModule.js
│       │     ├── DefaultResolve.js
│       │     ├── Loader.js
│       │     ├── ModuleJob.js
│       │     ├── ModuleMap.js
│       │     └── Translators.js
│       └── module.js
└── module.js
```

After (file names indicate the implemented module style):

```lib
├── ...
├── internal
│       ├── ...
│       ├── process
│       │     └── esm_loader.js
│       └── modules
│              ├── cjs
│              │     ├── helpers.js
│              │     └── loader.js
│              └── esm
│                    ├── CreateDynamicModule.js
│                    ├── DefaultResolve.js
│                    ├── Loader.js
│                    ├── ModuleJob.js
│                    ├── ModuleMap.js
│                    └── Translators.js
└── module.js
```

##### Checklist
<!-- Remove items that do not apply. For completed items, change [ ] to [x]. -->

- [x] `make -j4 test` (UNIX), or `vcbuild test` (Windows) passes
- [x] tests and/or benchmarks are included
- [x] commit message follows [commit guidelines](https://github.com/nodejs/node/blob/master/doc/guides/contributing/pull-requests.md#commit-message-guidelines)

joyeecheung commented:
cc @devsnek

addaleax commented on lib/module.js:
```diff
@@ -1,779 +1,3 @@
-
-// backwards compatibility
-Module.Module = Module;
+module.exports = require('internal/modules/cjsm');
```
I don’t think git recognizes this as a file move, so it won’t say anything about merge conflicts? That might be okay, we don’t have many PRs against this file open/to backport, I think…

joyeecheung replied:
@addaleax If we don't require that each commit must pass the test, we can delete the file first (so it will be recognized as a move) and then add the redirection later.

joyeecheung commented:
@devsnek I understand that the current `internal/modules/helper.js` can be renamed as `internal/modules/cjs/helper.js` (although my thinking is that we can add common helpers there so it does not need to be in any subdirectory. Things like `stripBOM` aren't really tied to any loader implementations).

Bikeshedding question: why does it need to be `cjs` instead of `cjsm`? CommonJS is not the equivalent of CommonJS modules, just like ECMAScript is not the equivalent of ECMAScript Modules (although we are not respecting the original specs in things like `assert` either).

> aside from that i worry about packages like trace possibly getting killed by at Module._compile (module.js) becoming at Module._compile (internal/modules/cjsm.js) but i haven't looked into that too deeply, so it might not be an issue

cc @AndreasMadsen although I don't think we should encourage user land modules to depend on our source positions..

devsnek commented:
@joyeecheung 
> Things like stripBOM aren't really tied to any loader implementations).

perhaps a better alternative would be moving that require building stuff out of the file then.

> Bikeshedding question: why does it need to be cjs instead of cjsm? 

everywhere else in the codebase that we shorten it we call it `cjs`, that's my only reasoning for it.

joyeecheung commented:
I also realize that there is the `internal/process/module.js` that should probably be renamed to `internal/process/esm_loader.js` since it's specific to setting up the ESM loader.

joyeecheung commented:
> everywhere else in the codebase that we shorten it we call it cjs, that's my only reasoning for it.

Looks like this has already been documented in the resolve hook formats..unfortunate, but probably better name the directory as `cjs` as well for consistency

joyeecheung commented:
> aside from that i worry about packages like trace.js clarify possibly getting killed by at Module._compile (module.js) becoming at Module._compile (internal/modules/cjsm.js) but i haven't looked into that too deeply, so it might not be an issue

I ran the test of clarify, it seems to be able to exclude all modules starting with `internal` so that should not be a problem.

joyeecheung commented:
@devsnek I've changed the structure, it now looks like this:

```
lib
├── ...
├── internal
│       ├── ...
│       └── modules
│              ├── cjs
│              │     ├── helpers.js
│              │     └── loader.js
│              └── esm
│                    ├── CreateDynamicModule.js
│                    ├── DefaultResolve.js
│                    ├── Loader.js
│                    ├── ModuleJob.js
│                    ├── ModuleMap.js
│                    └── Translators.js
└── module.js
```

PTAL, thanks!

joyeecheung commented:
@addaleax BTW, the file mode conflicts can be solved with this:

```
git format-patch <sha>~1...<sha>
# suppose the output is `0001-test.patch`
sed -i '' -E "s/lib\/module.js/lib\/internal\/modules\/cjs\/loader.js/" 0001-test.patch
git am -3 0001-.patch
```

If there are future backports (although given our stability concerns around the CJS loader implementation that should be rare), we just need to replace the `lib/module.js` in the email-formatted patch with `lib/internal/module/cjs/loader.js` and apply it with `git am -3`

joyeecheung commented:
@devsnek I think it's the variable that should be renamed to `helpers`? I believe they were named `internalModule` because the file was `internal/module`, not the other way around..

joyeecheung commented:
@devsnek Hmm, so the only `const internalModule = require('internal/modules/cjs/helpers');` left are the test `test-module-require-depth` and its fixtures..I changed to use destructuring instead since they only need the const values of `requireDepth`.

joyeecheung commented:
CI: https://ci.nodejs.org/job/node-test-pull-request/13546/

joyeecheung commented:
Rebased. New CI: https://ci.nodejs.org/job/node-test-pull-request/13574/

AndreasMadsen commented:
> aside from that i worry about packages like trace possibly getting killed by at Module._compile (module.js) becoming at Module._compile (internal/modules/cjsm.js) but i haven't looked into that too deeply, so it might not be an issue

> cc @AndreasMadsen although I don't think we should encourage user land modules to depend on our source positions.

It should still work. The `name.startsWith('internal')` check will match `internal/modules/cjsm.js`.

apapirovski commented on deps/node-inspect/lib/internal/inspect_repl.js:
```diff
@@ -86,7 +86,9 @@ function extractFunctionName(description) {
 const NATIVES = process.binding('natives');
 function isNativeUrl(url) {
-  return url.replace('.js', '') in NATIVES || url === 'bootstrap_node.js';
+  return url.replace('.js', '') in NATIVES ||
```
Should a PR be opened against https://github.com/nodejs/node-inspect before this lands so it doesn't get forgotten?

Also, strictly speaking, this change should either be in the same or preceding commit to the actual filename change (rather than after it), as otherwise node-inspect is broken.

(I just skimmed the PR so I could be totally off base.)

joyeecheung replied:
@apapirovski I just realized this is not necessary since

```
> Object.keys(process.binding('natives'))
[ 'internal/bootstrap/loaders',
  'internal/bootstrap/node',
  ...
```

joyeecheung commented:
Removed the unnecessary node-inspect change and rebased. CI: https://ci.nodejs.org/job/node-test-pull-request/13590/

@apapirovski PTAL, thanks

joyeecheung commented:
Rebased. I split the last commit as two, one moving `lib/module.js` to `lib/internal/modules/cjs/loader.js` without adding the redirection so that git recognize this as a file move and the `git blame` still works. Then add the redirection back so the test works. Therefore the last two commits must be landed together but I don't think that's going to be a problem since I'll backport those anyway.

CI: https://ci.nodejs.org/job/node-test-pull-request/13665/

joyeecheung commented:
Turned the NODE_DEBUG environment variable off in the `test-module-loading-globalpaths.js`, it makes the error message mixed up with debug output, looks like the subtle changes in the error stack now makes it break on osx10

<details>
<summary>See the mixed stderr</summary>

```
    MODULE 71864: looking for "/Users/iojs/build/workspace/node-test-commit-osx/nodes/osx1010/test/parallel/test-module-loading-globalpaths.js" in ["/Users/iojs/node-tmp/.tmp.1/home-no-pkg/.node_modules","/Users/iojs/node-tmp/.tmp.1/home-no-pkg/.node_libraries","/Users/iojs/node-tmp/.tmp.1/install/lib/node"]
    MODULE 71864: load "/Users/iojs/build/workspace/node-test-commit-osx/nodes/osx1010/test/parallel/test-module-loading-globalpaths.js" for module "."
    MODULE 71864: Module._load REQUEST ../common parent: .
    MODULE 71864: RELATIVE: requested: ../common set ID to: /Users/iojs/build/workspace/node-test-commit-osx/nodes/common from .
    MODULE 71864: looking for "/Users/iojs/build/workspace/node-test-commit-osx/nodes/common" in ["/Users/iojs/build/workspace/node-test-commit-osx/nodes/osx1010/test/parallel"]
    MODULE 71864: load "/Users/iojs/build/workspace/node-test-commit-osx/nodes/osx1010/test/common/index.js" for module "/Users/iojs/build/workspace/node-test-commit-osx/nodes/osx1010/test/common/index.js"
    MODULE 71864: Module._load REQUEST path parent: /Users/iojs/build/workspace/node-test-commit-osx/nodes/osx1010/test/common/index.js
    MODULE 71864: load native module path
    MODULE 71864: Module._load REQUEST fs parent: /Users/iojs/build/workspace/node-test-commit-osx/nodes/osx1010/test/common/index.js
    MODULE 71864: load native module fs
    MODULE 71864: Module._load REQUEST assert parent: /Users/iojs/build/workspace/node-test-commit-osx/nodes/osx1010/test/common/index.js
    MODULE 71864: load native module assert
    MODULE 71864: Module._load REQUEST os parent: /Users/iojs/build/workspace/node-test-commit-osx/nodes/osx1010/test/common/index.js
    MODULE 71864: load native module os
    MODULE 71864: Module._load REQUEST child_process parent: /Users/iojs/build/workspace/node-test-commit-osx/nodes/osx1010/test/common/index.js
    MODULE 71864: load native module child_process
    MODULE 71864: Module._load REQUEST stream parent: /Users/iojs/build/workspace/node-test-commit-osx/nodes/osx1010/test/common/index.js
    MODULE 71864: load native module stream
    MODULE 71864: Module._load REQUEST util parent: /Users/iojs/build/workspace/node-test-commit-osx/nodes/osx1010/test/common/index.js
    MODULE 71864: load native module util
    MODULE 71864: Module._load REQUEST ./fixtures parent: /Users/iojs/build/workspace/node-test-commit-osx/nodes/osx1010/test/common/index.js
    MODULE 71864: RELATIVE: requested: ./fixtures set ID to: /Users/iojs/build/workspace/node-test-commit-osx/nodes/osx1010/test/common/index.js/fixtures from /Users/iojs/build/workspace/node-test-commit-osx/nodes/osx1010/test/common/index.js
    MODULE 71864: looking for "/Users/iojs/build/workspace/node-test-commit-osx/nodes/osx1010/test/common/index.js/fixtures" in ["/Users/iojs/build/workspace/node-test-commit-osx/nodes/osx1010/test/common"]
    MODULE 71864: load "/Users/iojs/build/workspace/node-test-commit-osx/nodes/osx1010/test/common/fixtures.js" for module "/Users/iojs/build/workspace/node-test-commit-osx/nodes/osx1010/test/common/fixtures.js"
    MODULE 71864: Module._load REQUEST path parent: /Users/iojs/build/workspace/node-test-commit-osx/nodes/osx1010/test/common/fixtures.js
    MODULE 71864: load native module path
    MODULE 71864: Module._load REQUEST fs parent: /Users/iojs/build/workspace/node-test-commit-osx/nodes/osx1010/test/common/fixtures.js
    MODULE 71864: load native module fs
    MODULE 71864: Module._load REQUEST ./tmpdir parent: /Users/iojs/build/workspace/node-test-commit-osx/nodes/osx1010/test/common/index.js
    MODULE 71864: RELATIVE: requested: ./tmpdir set ID to: /Users/iojs/build/workspace/node-test-commit-osx/nodes/osx1010/test/common/index.js/tmpdir from /Users/iojs/build/workspace/node-test-commit-osx/nodes/osx1010/test/common/index.js
    MODULE 71864: looking for "/Users/iojs/build/workspace/node-test-commit-osx/nodes/osx1010/test/common/index.js/tmpdir" in ["/Users/iojs/build/workspace/node-test-commit-osx/nodes/osx1010/test/common"]
    MODULE 71864: load "/Users/iojs/build/workspace/node-test-commit-osx/nodes/osx1010/test/common/tmpdir.js" for module "/Users/iojs/build/workspace/node-test-commit-osx/nodes/osx1010/test/common/tmpdir.js"
    MODULE 71864: Module._load REQUEST fs parent: /Users/iojs/build/workspace/node-test-commit-osx/nodes/osx1010/test/common/tmpdir.js
    MODULE 71864: load native module fs
    MODULE 71864: Module._load REQUEST path parent: /Users/iojs/build/workspace/node-test-commit-osx/nodes/osx1010/test/common/tmpdir.js
    MODULE 71864: load native module path
    MODULE 71864: Module._load REQUEST ../common/fixtures parent: .
    MODULE 71864: RELATIVE: requested: ../common/fixtures set ID to: /Users/iojs/build/workspace/node-test-commit-osx/nodes/common/fixtures from .
    MODULE 71864: looking for "/Users/iojs/build/workspace/node-test-commit-osx/nodes/common/fixtures" in ["/Users/iojs/build/workspace/node-test-commit-osx/nodes/osx1010/test/parallel"]
    MODULE 71864: Module._load REQUEST assert parent: .
    MODULE 71864: load native module assert
    MODULE 71864: Module._load REQUEST path parent: .
    MODULE 71864: load native module path
    MODULE 71864: Module._load REQUEST fs parent: .
    MODULE 71864: load native module fs
    MODULE 71864: Module._load REQUEST child_process parent: .
    MODULE 71864: load native module child_process
    MODULE 71864: Module._load REQUEST ../common/shared-lib-util parent: .
    MODULE 71864: RELATIVE: requested: ../common/shared-lib-util set ID to: /Users/iojs/build/workspace/node-test-commit-osx/nodes/common/shared-lib-util from .
    MODULE 71864: looking for "/Users/iojs/build/workspace/node-test-commit-osx/nodes/common/shared-lib-util" in ["/Users/iojs/build/workspace/node-test-commit-osx/nodes/osx1010/test/parallel"]
    MODULE 71864: load "/Users/iojs/build/workspace/node-test-commit-osx/nodes/osx1010/test/common/shared-lib-util.js" for module "/Users/iojs/build/workspace/node-test-commit-osx/nodes/osx1010/test/common/shared-lib-util.js"
    MODULE 71864: Module._load REQUEST ../common parent: /Users/iojs/build/workspace/node-test-commit-osx/nodes/osx1010/test/common/shared-lib-util.js
    MODULE 71864: RELATIVE: requested: ../common set ID to: /Users/iojs/build/workspace/node-test-commit-osx/nodes/osx1010/test/common from /Users/iojs/build/workspace/node-test-commit-osx/nodes/osx1010/test/common/shared-lib-util.js
    MODULE 71864: looking for "/Users/iojs/build/workspace/node-test-commit-osx/nodes/osx1010/test/common" in ["/Users/iojs/build/workspace/node-test-commit-osx/nodes/osx1010/test/common"]
    MODULE 71864: Module._load REQUEST path parent: /Users/iojs/build/workspace/node-test-commit-osx/nodes/osx1010/test/common/shared-lib-util.js
    MODULE 71864: load native module path
    MODULE 71864: Module._load REQUEST foo parent: .
    MODULE 71864: looking for "foo" in ["/Users/iojs/build/workspace/node-test-commit-osx/nodes/osx1010/test/parallel/node_modules","/Users/iojs/build/workspace/node-test-commit-osx/nodes/osx1010/test/node_modules","/Users/iojs/build/workspace/node-test-commit-osx/nodes/osx1010/node_modules","/Users/iojs/build/workspace/node-test-commit-osx/nodes/node_modules","/Users/iojs/build/workspace/node-test-commit-osx/node_modules","/Users/iojs/build/workspace/node_modules","/Users/iojs/build/node_modules","/Users/iojs/node_modules","/Users/node_modules","/node_modules","/Users/iojs/node-tmp/.tmp.1/home-no-pkg/.node_modules","/Users/iojs/node-tmp/.tmp.1/home-no-pkg/.node_libraries","/Users/iojs/node-tmp/.tmp.1/install/lib/node"]
    assert.js:465
        throw actual;
        ^
    
    Error: Command failed: /Users/iojs/node-tmp/.tmp.1/install/bin/node /Users/iojs/build/workspace/node-test-commit-osx/nodes/osx1010/test/parallel/test-module-loading-globalpaths.js child
    MODULE 71864: looking for "/Users/iojs/build/workspace/node-test-commit-osx/nodes/osx1010/test/parallel/test-module-loading-globalpaths.js" in ["/Users/iojs/node-tmp/.tmp.1/home-no-pkg/.node_modules","/Users/iojs/node-tmp/.tmp.1/home-no-pkg/.node_libraries","/Users/iojs/node-tmp/.tmp.1/install/lib/node"]
    MODULE 71864: load "/Users/iojs/build/workspace/node-test-commit-osx/nodes/osx1010/test/parallel/test-module-loading-globalpaths.js" for module "."
    MODULE 71864: Module._load REQUEST ../common parent: .
    MODULE 71864: RELATIVE: requested: ../common set ID to: /Users/iojs/build/workspace/node-test-commit-osx/nodes/common from .
    MODULE 71864: looking for "/Users/iojs/build/workspace/node-test-commit-osx/nodes/common" in ["/Users/iojs/build/workspace/node-test-commit-osx/nodes/osx1010/test/parallel"]
    MODULE 71864: load "/Users/iojs/build/workspace/node-test-commit-osx/nodes/osx1010/test/common/index.js" for module "/Users/iojs/build/workspace/node-test-commit-osx/nodes/osx1010/test/common/index.js"
    MODULE 71864: Module._load REQUEST path parent: /Users/iojs/build/workspace/node-test-commit-osx/nodes/osx1010/test/common/index.js
    MODULE 71864: load native module path
    MODULE 71864: Module._load REQUEST fs parent: /Users/iojs/build/workspace/node-test-commit-osx/nodes/osx1010/test/common/index.js
    MODULE 71864: load native module fs
    MODULE 71864: Module._load REQUEST assert parent: /Users/iojs/build/workspace/node-test-commit-osx/nodes/osx1010/test/common/index.js
    MODULE 71864: load native module assert
    MODULE 71864: Module._load REQUEST os parent: /Users/iojs/build/workspace/node-test-commit-osx/nodes/osx1010/test/common/index.js
    MODULE 71864: load native module os
    MODULE 71864: Module._load REQUEST child_process parent: /Users/iojs/build/workspace/node-test-commit-osx/nodes/osx1010/test/common/index.js
    MODULE 71864: load native module child_process
    MODULE 71864: Module._load REQUEST stream parent: /Users/iojs/build/workspace/node-test-commit-osx/nodes/osx1010/test/common/index.js
    MODULE 71864: load native module stream
    MODULE 71864: Module._load REQUEST util parent: /Users/iojs/build/workspace/node-test-commit-osx/nodes/osx1010/test/common/index.js
    MODULE 71864: load native module util
    MODULE 71864: Module._load REQUEST ./fixtures parent: /Users/iojs/build/workspace/node-test-commit-osx/nodes/osx1010/test/common/index.js
    MODULE 71864: RELATIVE: requested: ./fixtures set ID to: /Users/iojs/build/workspace/node-test-commit-osx/nodes/osx1010/test/common/index.js/fixtures from /Users/iojs/build/workspace/node-test-commit-osx/nodes/osx1010/test/common/index.js
    MODULE 71864: looking for "/Users/iojs/build/workspace/node-test-commit-osx/nodes/osx1010/test/common/index.js/fixtures" in ["/Users/iojs/build/workspace/node-test-commit-osx/nodes/osx1010/test/common"]
    MODULE 71864: load "/Users/iojs/build/workspace/node-test-commit-osx/nodes/osx1010/test/common/fixtures.js" for module "/Users/iojs/build/workspace/node-test-commit-osx/nodes/osx1010/test/common/fixtures.js"
    MODULE 71864: Module._load REQUEST path parent: /Users/iojs/build/workspace/node-test-commit-osx/nodes/osx1010/test/common/fixtures.js
    MODULE 71864: load native module path
    MODULE 71864: Module._load REQUEST fs parent: /Users/iojs/build/workspace/node-test-commit-osx/nodes/osx1010/test/common/fixtures.js
    MODULE 71864: load native module fs
    MODULE 71864: Module._load REQUEST ./tmpdir parent: /Users/iojs/build/workspace/node-test-commit-osx/nodes/osx1010/test/common/index.js
    MODULE 71864: RELATIVE: requested: ./tmpdir set ID to: /Users/iojs/build/workspace/node-test-commit-osx/nodes/osx1010/test/common/index.js/tmpdir from /Users/iojs/build/workspace/node-test-commit-osx/nodes/osx1010/test/common/index.js
    MODULE 71864: looking for "/Users/iojs/build/workspace/node-test-commit-osx/nodes/osx1010/test/common/index.js/tmpdir" in ["/Users/iojs/build/workspace/node-test-commit-osx/nodes/osx1010/test/common"]
    MODULE 71864: load "/Users/iojs/build/workspace/node-test-commit-osx/nodes/osx1010/test/common/tmpdir.js" for module "/Users/iojs/build/workspace/node-test-commit-osx/nodes/osx1010/test/common/tmpdir.js"
    MODULE 71864: Module._load REQUEST fs parent: /Users/iojs/build/workspace/node-test-commit-osx/nodes/osx1010/test/common/tmpdir.js
    MODULE 71864: load native module fs
    MODULE 71864: Module._load REQUEST path parent: /Users/iojs/build/workspace/node-test-commit-osx/nodes/osx1010/test/common/tmpdir.js
    MODULE 71864: load native module path
    MODULE 71864: Module._load REQUEST ../common/fixtures parent: .
    MODULE 71864: RELATIVE: requested: ../common/fixtures set ID to: /Users/iojs/build/workspace/node-test-commit-osx/nodes/common/fixtures from .
    MODULE 71864: looking for "/Users/iojs/build/workspace/node-test-commit-osx/nodes/common/fixtures" in ["/Users/iojs/build/workspace/node-test-commit-osx/nodes/osx1010/test/parallel"]
    MODULE 71864: Module._load REQUEST assert parent: .
    MODULE 71864: load native module assert
    MODULE 71864: Module._load REQUEST path parent: .
    MODULE 71864: load native module path
    MODULE 71864: Module._load REQUEST fs parent: .
    MODULE 71864: load native module fs
    MODULE 71864: Module._load REQUEST child_process parent: .
    MODULE 71864: load native module child_process
    MODULE 71864: Module._load REQUEST ../common/shared-lib-util parent: .
    MODULE 71864: RELATIVE: requested: ../common/shared-lib-util set ID to: /Users/iojs/build/workspace/node-test-commit-osx/nodes/common/shared-lib-util from .
    MODULE 71864: looking for "/Users/iojs/build/workspace/node-test-commit-osx/nodes/common/shared-lib-util" in ["/Users/iojs/build/workspace/node-test-commit-osx/nodes/osx1010/test/parallel"]
    MODULE 71864: load "/Users/iojs/build/workspace/node-test-commit-osx/nodes/osx1010/test/common/shared-lib-util.js" for module "/Users/iojs/build/workspace/node-test-commit-osx/nodes/osx1010/test/common/shared-lib-util.js"
    MODULE 71864: Module._load REQUEST ../common parent: /Users/iojs/build/workspace/node-test-commit-osx/nodes/osx1010/test/common/shared-lib-util.js
    MODULE 71864: RELATIVE: requested: ../common set ID to: /Users/iojs/build/workspace/node-test-commit-osx/nodes/osx1010/test/common from /Users/iojs/build/workspace/node-test-commit-osx/nodes/osx1010/test/common/shared-lib-util.js
    MODULE 71864: looking for "/Users/iojs/build/workspace/node-test-commit-osx/nodes/osx1010/test/common" in ["/Users/iojs/build/workspace/node-test-commit-osx/nodes/osx1010/test/common"]
    MODULE 71864: Module._load REQUEST path parent: /Users/iojs/build/workspace/node-test-commit-osx/nodes/osx1010/test/common/shared-lib-util.js
    MODULE 71864: load native module path
    MODULE 71864: Module._load REQUEST foo parent: .
    MODULE 71864: looking for "foo" in ["/Users/iojs/build/workspace/node-test-commit-osx/nodes/osx1010/test/parallel/node_modules","/Users/iojs/build/workspace/node-test-commit-osx/nodes/osx1010/test/node_modules","/Users/iojs/build/workspace/node-test-commit-osx/nodes/osx1010/node_modules","/Users/iojs/build/workspace/node-test-commit-osx/nodes/node_modules","/Users/iojs/build/workspace/node-test-commit-osx/node_modules","/Users/iojs/build/workspace/node_modules","/Users/iojs/build/node_modules","/Users/iojs/node_modules","/Users/node_modules","/node_modules","/Users/iojs/node-tmp/.tmp.1/home-no-pkg/.node_modules","/Users/iojs/node-tmp/.tmp.1/home-no-pkg/.node_libraries","/Users/iojs/node-tmp/.tmp.1/install/lib/node"]
    
        at checkExecSyncError (child_process.js:591:11)
        at Object.execFileSync (child_process.js:609:13)
        at assert.throws (/Users/iojs/build/workspace/node-test-commit-osx/nodes/osx1010/test/parallel/test-module-loading-globalpaths.js:55:21)
        at getActual (assert.js:421:5)
        at Function.throws (assert.js:493:24)
        at Object.<anonymous> (/Users/iojs/build/workspace/node-test-commit-osx/nodes/osx1010/test/parallel/test-module-loading-globalpaths.js:53:10)
        at Module._compile (internal/modules/cjs/loader.js:676:30)
        at Object.Module._extensions..js (internal/modules/cjs/loader.js:687:10)
        at Module.load (internal/modules/cjs/loader.js:587:32)
        at tryModuleLoad (internal/modules/cjs/loader.js:527:12)
```
</details>

CI: https://ci.nodejs.org/job/node-test-pull-request/13676/

joyeecheung commented:
CI is green. @nodejs/collaborators going to land this tomorrow if no objections.

mcollina commented:
I'm wondering if this touches any of the undocumented APIs. If it does, then it should probably be semver-major.

A citgm run should be good anyway.

joyeecheung commented:
> I'm wondering if this touches any of the undocumented APIs. If it does, then it should probably be semver-major.

Not to my knowledge, since this PR just moves internal files around.

Just to be safe:

CITGM against master: https://ci.nodejs.org/view/Node.js-citgm/job/citgm-smoker/1332/
CITGM against this PR: https://ci.nodejs.org/view/Node.js-citgm/job/citgm-smoker/1332/

joyeecheung commented:
No relavent new errors in CITGM AFAICT. Landing..

joyeecheung commented:
Landed in 1ac1424476...54de79b16e, thanks!

MylesBorins commented:
Should this be backported to `v9.x-staging`? If yes please follow [the guide](https://github.com/nodejs/node/blob/master/doc/guides/backporting-to-release-lines.md) and raise a backport PR, if not let me know or add the `dont-land-on` label.

joyeecheung commented:
@MylesBorins Backport opened in https://github.com/nodejs/node/pull/19374 , somehow I forgot to reference the PR from there so it did not show up in this thread
